app/helpers/application_helper.rb

```diff
@@ -67,14 +67,29 @@ def navigation_link_is_for_an_enabled_feature?(link:)
   #
   # rubocop:disable Rails/HelperInstanceVariable
   def view_class
-    if @podcast_episode_show # custom due to edge cases
-      "stories stories-show podcast_episodes-show"
-    elsif @story_show
-      "stories stories-show"
-    else
-      "#{controller_name} #{current_page}"
-    end
+    base_classes = if @podcast_episode_show # custom due to edge cases
+                     "stories stories-show podcast_episodes-show"
+                   elsif @story_show
+                     "stories stories-show"
+                   else
+                     "#{controller_name} #{current_page}"
+                   end
+    base_classes += article_view_classes if @article&.class&.name&.start_with?("Article") # Article or ArticleDecorator
+    base_classes += page_view_classes if @page&.class&.name&.start_with?("Page")
```

Comment:
nit: a more succinct approach
```suggestion
    base_classes += article_view_classes if @article.instance_of?(Article) || @article.instance_of?(ArticleDecorator)
    base_classes += page_view_classes if @page.instance_of?(Page)
```